### PR TITLE
Add microstate accounting collector

### PR DESCRIPTION
**Pull request by seancribbs**

### CODE REVIEW

File: doc/prometheus_vm_microstates_collector.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+
+
+# Module prometheus_vm_microstates_collector #
+* [Description](#description)
+
+Collects statistics about the microstate accounting of Erlang VM threads using
+[msacc](http://erlang.org/doc/man/msacc.md).
```
Comment (tsloughter) [inline doc/prometheus_vm_microstates_collector.md]:
```suggestion
[msacc](http://erlang.org/doc/man/msacc.html).
```

File: doc/prometheus_vm_microstates_collector.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+__Behaviours:__ [`prometheus_collector`](prometheus_collector.md).
+
+<a name="description"></a>
+
+## Description ##
+
+Refer to
+[
+erlang:statistics(microstate_accounting)
+](http://erlang.org/doc/man/erlang.md#statistics_microstate_accounting) for more information about thread types and microstates.
```
Comment (tsloughter) [inline doc/prometheus_vm_microstates_collector.md]:
```suggestion
](http://erlang.org/doc/man/erlang.html#statistics_microstate_accounting) for more information about thread types and microstates.
```

I don't know where you got these `.md` links from but they return 404 for me but `.html` works :)

File: doc/prometheus_vm_microstates_collector.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+* `erlang_vm_thread_microstates_microseconds{thread_type="async|aux|dirty_io_scheduler|dirty_cpu_scheduler|poll|scheduler", thread_state="alloc|aux|bif|busy_wait|check_io|emulator|ets|gc|gc_fullsweep|nif|other|port|send|sleep|timers", thread_id=ID}`<br />
+Type: counter<br />
+The time spent in the labeled microstate since collection began for the labeled VM thread.
+
+
+
+### <a name="Configuration">Configuration</a> ###
+
+The [time
+unit](http://erlang.org/doc/man/erlang.md#type-time_unit) for this collector can be configured via
```
Comment (tsloughter) [inline doc/prometheus_vm_microstates_collector.md]:
```suggestion
unit](http://erlang.org/doc/man/erlang.html#type-time_unit) for this collector can be configured via
```

File: doc/prometheus_vm_microstates_collector.md (hunk 1 above)
Comment (seancribbs) [reply doc/prometheus_vm_microstates_collector.md]: 🤔 That was the doclet changing it, I directly pasted the URL into the source.

**Comment by deadtrickster** [comment]
Hey, Thanks!
Are we sure it should enabled by default?

**Comment by gerhard** [comment]
Isn't this a dupe of #98 ?

**Comment by seancribbs** [comment]
@gerhard You're right, I guess I didn't notice it before submitting.

**Comment by seancribbs** [comment]
> Hey, Thanks!
> Are we sure it should enabled by default?

@deadtrickster No, I'm not sure. How does everyone feel about that?

**Comment by seancribbs** [comment]
Closing this since #98 was merged.

**Pull request closed without merging**